src/lib.rs

```diff
@@ -22,214 +41,534 @@ pub enum SdSpecVersion {
+            2 => BusWidth::Four,
+            _ => BusWidth::Unknown,
+        }
+    }
+    /// Is the card currently in the secured mode
+    pub fn secure_mode(&self) -> bool {
+        self.inner[15] & 0x2000_0000 != 0
+    }
+    /// SD Memory Card type (ROM, OTP, etc)
+    pub fn sd_memory_card_type(&self) -> u16 {
+        self.inner[15] as u16
+    }
+    /// SDHC / SDXC: Capacity of Protected Area in bytes
+    pub fn protected_area_size(&self) -> u32 {
+        self.inner[14]
+    }
+    /// Speed Class
+    pub fn speed_class(&self) -> u8 {
+        (self.inner[13] >> 24) as u8
+    }
+    /// "Performance Move" indicator in 1 MB/s units
+    pub fn move_performance(&self) -> u8 {
+        (self.inner[13] >> 16) as u8
+    }
+    /// Allocation Unit (AU) size. Lookup in PLSS v7_10 Table 4-47
+    pub fn allocation_unit_size(&self) -> u8 {
+        (self.inner[13] >> 12) as u8 & 0xF
+    }
+    /// Indicates N_Erase, in units of AU
+    pub fn erase_size(&self) -> u16 {
+        (self.inner[13] & 0xFF) as u16 | ((self.inner[12] >> 24) & 0xFF) as u16
+    }
+    /// Indicates T_Erase / Erase Timeout (s)
+    pub fn erase_timeout(&self) -> u8 {
+        (self.inner[12] >> 18) as u8 & 0x3F
+    }
+    /// Video speed class
+    pub fn video_speed_class(&self) -> u8 {
+        (self.inner[11] & 0xFF) as u8
+    }
+    /// Application Performance Class
+    pub fn app_perf_class(&self) -> u8 {
+        (self.inner[9] >> 16) as u8 & 0xF
+    }
+    /// Discard Support
+    pub fn discard_support(&self) -> bool {
+        self.inner[8] & 0x0200_0000 != 0
+    }
+}
+impl fmt::Debug for SDStatus {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("SD Status")
+            .field("Bus Width", &self.bus_width())
+            .field("Secured Mode", &self.secure_mode())
+            .field("SD Memory Card Type", &self.sd_memory_card_type())
+            .field("Protected Area Size (B)", &self.protected_area_size())
+            .field("Speed Class", &self.speed_class())
+            .field("Video Speed Class", &self.video_speed_class())
+            .field("Application Performance Class", &self.app_perf_class())
+            .field("Move Performance (MB/s)", &self.move_performance())
+            .field("AU Size", &self.allocation_unit_size())
+            .field("Erase Size (units of AU)", &self.erase_size())
+            .field("Erase Timeout (s)", &self.erase_timeout())
+            .field("Discard Support", &self.discard_support())
+            .finish()
+    }
+}
+
+/// Relative Card Address (RCA)
+///
+/// R6
+#[derive(Copy, Clone, Default)]
+pub struct RCA(u32);
+impl From<u32> for RCA {
+    fn from(word: u32) -> Self {
+        Self(word)
+    }
+}
+impl RCA {
+    /// Address of card
+    pub fn address(&self) -> u16 {
+        (self.0 >> 16) as u16
+    }
+    /// Status
+    pub fn status(&self) -> u16 {
+        self.0 as u16
+    }
+}
+
+/// Card interface condition (R7)
+///
+/// R6
```

Comment:
The Physical Layer Simplified Specification calls this response R7, it's not mentioned in the stm32f4 reference manual. So we should probably just remove the R6 here?

Reply:
Good spot - it's just a copy+paste error